tests/integration/conftest.py

```diff
@@ -985,12 +1036,102 @@ def assert_qtest_property(model, prop_name, exp_prop_value, promiscuous=True):
+    qTest admin of the qTest project under test.
+
+    Enabling 'promiscuous' mode (default) will attempt to match the property value against either the 'field_value' or
+    'field_value_name' attributes of the model. If 'promiscuous' mode is disabled then validation of the expected
+    property value will only be made against the 'field_value' attribute of the model.
+
+    Args:
+        model (object): Any model from the 'swagger_client.models' namespace.
+        prop_name (str): Target property name to use for value validation.
+        exp_prop_value (str): Expected value for the given property name.
+        promiscuous (bool): Flag for indicating whether to run assertion in 'promiscuous' mode or not. (See above)
+
+    Raises:
+        AssertionError: Property name does not exist or property value does not match expected value.
+    """
+
+    actual_values = get_qtest_property(model, prop_name)
+
     if promiscuous:
         assert exp_prop_value in actual_values
     else:
         assert actual_values[0] == exp_prop_value
 
 
+# noinspection PyUnresolvedReferences
+@pytest.helpers.register
+def assert_qtest_property_match(model, prop_name, exp_prop_regex, regex_flags=0, promiscuous=True):
+    """Assert that a qTest swagger_client model has a property that matches a given regular expression. This assert will
+    intelligently search both standard qTest swagger_client model attributes as well as custom fields set by the
+    qTest admin of the qTest project under test.
+
+    NOTE: Regex is run in 'match' mode with this helper!
+
+    Enabling 'promiscuous' mode (default) will attempt to match the property value against either the 'field_value' or
+    'field_value_name' attributes of the model. If 'promiscuous' mode is disabled then validation of the expected
+    property value will only be made against the 'field_value' attribute of the model.
+
+    Args:
+        model (object): Any model from the 'swagger_client.models' namespace.
+        prop_name (str): Target property name to use for value validation.
+        exp_prop_regex (str): Regular expression pattern that property value is expected to match.
+        regex_flags (int): Flags to pass to the regex evaluator to alter pattern matching behavior.
+        promiscuous (bool): Flag for indicating whether to run assertion in 'promiscuous' mode or not. (See above)
+
+    Raises:
+        AssertionError: Property name does not exist or property value failed regular expression match.
+    """
+
+    actual_values = get_qtest_property(model, prop_name)
+
+    if promiscuous:
+        assert any([re.match(exp_prop_regex, value, regex_flags) for value in actual_values if type(value) is str])
+    else:
+        assert re.match(exp_prop_regex, actual_values[0], regex_flags)
+
+
+# noinspection PyUnresolvedReferences
+@pytest.helpers.register
+def assert_qtest_property_search(model, prop_name, exp_prop_regex, regex_flags=0, promiscuous=True):
```

Comment:
Turns out that extra characters can be added depending on the format of the JUnitXML. So needed to have a way to run regex pattern matching.